### DISCUSSION TITLE
Run Dialog improvements

### DISCUSSION THREAD

**Discussion by alttabber** [linuxmint/discussions#142] [Ideas]
The run dialog hasn't seen a feature upgrade since forever, and in its current state is clunky and unintuitive, something that only nerds would ever dream of using. I'd say it's about time it gets an overhaul.

After Dmenu, Rofi, KRunner, Xfce's runner, Ulauncher, Launchy, and many other, I'd say the general consensus is that a runner should be a real time search, which Cinnamon already does in the menu, but not in the run dialog.

I think the run dialog should become a live search of .desktop applications, additional behaviors could be selected with a menu or with prefixes (ex. `$ command` to run an arbitrary command, `f /path/to/file` to open a file/directory). Going crazy about it, it might even get extensions.

Of course, the search should be pretty, with icons and animations, on par with the rest of the Cinnamon desktop.

Last, but not least, even if everything else doesn't get implemented, it should have an help window. As it is now, the UI isn't self-explanatory and there is no trivially available documentation on what it can do.

**Comment by kacperpaczos** [comment]
Has Linux Mint a run dialog? Please attach a screenshot of this, because i don't know this feature.

**Reply by alttabber** [reply]
It's bound to Alt+F2 by default.

**Reply by alttabber** [reply]
Shot on Mint 21.2's live CD in a KVM.
![Screenshot_1](https://github.com/linuxmint/discussions/assets/134143997/e6f55e47-7789-4e2d-ac1f-ddd6a4959f03)
![Screenshot_0](https://github.com/linuxmint/discussions/assets/134143997/6033b93a-0f06-4e9c-ab5e-602c039223ca)
